atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/ValueCacheSnapshotImpl.java

```diff
@@ -30,21 +32,31 @@ public interface ValueCacheSnapshotImpl extends ValueCacheSnapshot {
 
     Set<TableReference> lockWatchEnabledTables();
 
+    // The tables from these row refs may or may not be listed in lockWatchEnabledTables
+    Set<RowReference> lockWatchEnabledRows();
```

Comment:
note that at present, we **do** need to keep a set of enabled rows here as a source of truth, although this is unfortunately currently coupled with the "value cache snapshot". Work to decouple these concerns would be part of a future PR, **after** adding tests to show that row-level watches (explicitly without caching) are supported.